src/engine/renderer/tr_image.cpp

```diff
@@ -1944,6 +1944,9 @@ byte *R_Resize( byte *in, int width, int height, int width2, int height2 )
 
 	out = (byte*) ri.Z_Malloc( width2 * height2 * 4 );
 
+	ResampleTexture( (unsigned int*) in, width, height, (unsigned int*) out, width2, height2, false );
```

Comment:
@slipher I tried to reuse `ResampleTexture` since I see no reason it would not work for upscaling too.
It works but I had to hack some casts, basically the cubemap faces are bytes (`unsigned char`) while `ResampleTexture` expects `unsigned int`).

It looks like that `ResampleTexture` function was never used before, so perhaps we can fix it…

This is with `R_Resize`:

 [![skybox](https://dl.illwieckz.net/b/daemon/bugs/noncube-skybox/unvanquished_2019-03-06_200229_000.jpg)](https://dl.illwieckz.net/b/daemon/bugs/noncube-skybox/unvanquished_2019-03-06_200229_000.jpg)

This is with `ResampleTexture`:

[![skybox](https://dl.illwieckz.net/b/daemon/bugs/noncube-skybox/unvanquished_2019-03-09_202118_000.jpg)](https://dl.illwieckz.net/b/daemon/bugs/noncube-skybox/unvanquished_2019-03-09_202118_000.jpg)

It looks like we may drop one of them.

Reply:
There is one place where `ResampleTexture` was used, but it used the cast hack too, so we can rewrite the `ResampleTexture` definition…

Reply:
@slipher I have not managed to get valid results by redefining `ResampleTexture`, so I had to kept the ugly casts.

In any way I dropped the custom resizing code to rely on `ResampleTexture`. I kept `R_Resize` as a wrapper to make in-place resizing easier.

Reply:
OK with me. The difference is that ResampleTexture takes an evenly weighted average of 4 pixels. Maybe the algorithm doesn't make that much sense for scaling up, but of course it doesn't matter since it's only used as a workaround for assets which by rights shouldn't work at all.

Reply:
Thanks for the review!